src/mosaic.F90

```diff
@@ -527,6 +588,59 @@ end subroutine aerosol_optical
 
   end subroutine mosaic_aero_optical_init
 
+!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!
+
+  subroutine mosaic_compute_single_aero_optical(env_state, aero_data, &
```

Comment:
Yes, this is for single wavelength. The wavelength, actually all wavelengths, are actually unknown to PartMC as it lives in MOSAIC. We just know that it is 550 nm. We should in general handle this by outputting the wavelengths by getting them from MOSAIC. For this subroutine, it's currently the third wavelength in the array - hardcoded over on the MOSAIC side (which is the 550 nm in my version of MOSAIC).

I think we should clean this up. There's certainly a more clever solution to cleaning up this code in general. ` aerosol_optical` and `aerosol_optical_single_wavelength` are nearly identical (the latter just has a wavelength loop removed. This could be solved by passing an argument to `aerosol_optical` to determine what wavelength(s) it computes on.

Reply:
That definitely seems like a sensible cleanup to do.

Reply:
Done in https://github.com/compdyn/partmc/pull/168/commits/33cafdb6d65403bb0190efa50f6290f2e2502745

Reply:
Refactored my MOSAIC code to remove `aerosol_optical_single_wavelength` and added an optional argument to `aerosol_optical` to pick a wavelength index, which is selected in PartMC and no longer hardcoded in MOSAIC to be 3 (which is currently 550 nm).

https://github.com/compdyn/partmc/pull/168/commits/3a1abb3e89194d7b40d44897df8c649ae9910223